### ISSUE TITLE
ValueError: Invalid server scheme: socks4

### ISSUE THREAD

**Issue by Avnsx** [wkeeling/selenium-wire#256]
Hey @wkeeling , how come I can't use socks4 proxies but only socks5 and https proxies?

```python
import re
import undetected_chromedriver as uc
from seleniumwire import webdriver

chrome_options = webdriver.ChromeOptions()
chrome_options.headless = False
seleniumwire_options = {'proxy': {'https': 'socks4://139.162.176.201:9200',}}
driver = uc.Chrome(options=chrome_options, seleniumwire_options=seleniumwire_options)

driver.get('https://api.myip.com/')

////////////////////////////////////////////////////

Traceback (most recent call last):
  File "C:\Users\me\Downloads\folder\mycode.py", line 8, in <module>
    driver = uc.Chrome(options=chrome_options, seleniumwire_options=seleniumwire_options)
  File "C:\Users\me\AppData\Local\Programs\Python\Python39\lib\site-packages\undetected_chromedriver\__init__.py", line 53, in __new__
    instance.__init__(*args, **kwargs)
  File "C:\Users\me\AppData\Local\Programs\Python\Python39\lib\site-packages\seleniumwire\webdriver.py", line 85, in __init__
    self.proxy = backend.create(
  File "C:\Users\me\AppData\Local\Programs\Python\Python39\lib\site-packages\seleniumwire\backend.py", line 35, in create
    proxy = MitmProxy(addr, port, options)
  File "C:\Users\me\AppData\Local\Programs\Python\Python39\lib\site-packages\seleniumwire\server.py", line 58, in __init__
    mitmproxy_opts.update(
  File "C:\Users\me\AppData\Local\Programs\Python\Python39\lib\site-packages\seleniumwire\thirdparty\mitmproxy\optmanager.py", line 223, in update
    u = self.update_known(**kwargs)
  File "C:\Users\me\AppData\Local\Programs\Python\Python39\lib\site-packages\seleniumwire\thirdparty\mitmproxy\optmanager.py", line 215, in update_known
    self.changed.send(self, updated=updated)
  File "C:\Users\me\AppData\Local\Programs\Python\Python39\lib\site-packages\blinker\base.py", line 266, in send
    return [(receiver, receiver(sender, **kwargs))
  File "C:\Users\me\AppData\Local\Programs\Python\Python39\lib\site-packages\blinker\base.py", line 266, in <listcomp>
    return [(receiver, receiver(sender, **kwargs))
  File "C:\Users\me\AppData\Local\Programs\Python\Python39\lib\site-packages\seleniumwire\thirdparty\mitmproxy\server\config.py", line 90, in configure
    _, spec = server_spec.parse_with_mode(options.mode)
  File "C:\Users\me\AppData\Local\Programs\Python\Python39\lib\site-packages\seleniumwire\thirdparty\mitmproxy\net\server_spec.py", line 80, in parse_with_mode
    return mode, parse(server_spec)
  File "C:\Users\me\AppData\Local\Programs\Python\Python39\lib\site-packages\seleniumwire\thirdparty\mitmproxy\net\server_spec.py", line 47, in parse
    raise ValueError("Invalid server scheme: {}".format(scheme))
ValueError: Invalid server scheme: socks4

**Comment by wkeeling** [comment]
Good catch @Avnsx. The refactoring that took place in v4.0.0 omitted socks4 from the list of permitted schemes, causing the validation error above. Socks 4 seems to be less popular than socks 5, so the problem hadn't shown up until you just found it. 

Now fixed in v4.2.3 - are you able to confirm OK?

**Comment by Avnsx** [comment]
> Good catch @Avnsx. The refactoring that took place in v4.0.0 omitted socks4 from the list of permitted schemes, causing the validation error above. Socks 4 seems to be less popular than socks 5, so the problem hadn't shown up until you just found it.
> 
> Now fixed in v4.2.3 - are you able to confirm OK?

Yes, that fixed it. Thanks alot for the quick response and fix. 👍
